### ISSUE TITLE
Allow output in multiple formats at the same time

### ISSUE THREAD

**Issue by xixixao** [microsoft/pyright#10955]
We're currently running pyright in CI twice, to get a machine-readable report output (shown in GitLab UI) and human-readable log (show in the CI job log), like this:

```sh
pyright . --outputjson > report_raw.json ||
pyright-to-gitlab-ci --src report_raw.json --output report.json --base_path . && 
pyright .
```

This slows down failed jobs. We would love to be able to get both outputs without rerunning typechecking, like:

```
pyright . --output json:report_raw.json --output text:stdout
```

There are some edge cases with this syntax, but they're probably ok (like you cannot output to a file called `stdout`). It could be some other syntax, and it can definitely be backwards compatible.

Thanks for considering this!

**Comment by erictraut** [comment]
With the machine-readable output, you can easily write a script to format the output in any manner you want — including a format that's human-readable. For that reason, I don't think it makes sense for pyright to offer a feature to output both.

**Comment by xixixao** [comment]
That would mean reengineering how pyright normally outputs, to stay close to it. Not sure that's easy to do. But understand this is low priority.
